test/unit/ButtonTokenWethRouter.ts

```diff
@@ -0,0 +1,156 @@
+import { ethers, waffle } from 'hardhat'
+import { Contract, Signer, BigNumber } from 'ethers'
+import { expect } from 'chai'
+const { loadFixture } = waffle
+
+const PRICE_DECIMALS = 8
+const NAME = 'Button Bitcoin'
+const SYMBOL = 'BTN-BTC'
+
+interface TestContext {
+  accounts: Signer[]
+  buttonToken: Contract
+  mockOracle: Contract
+  weth: Contract
+  router: Contract
+}
+
+const toOracleValue = (v: string): BigNumber =>
+  ethers.utils.parseUnits(v, PRICE_DECIMALS)
+
+async function fixture(): Promise<TestContext> {
+  const accounts = await ethers.getSigners()
+
+  const wethFactory = await ethers.getContractFactory('WETH9')
+  const weth = await wethFactory.deploy()
+
+  const oracleFactory = await ethers.getContractFactory('MockOracle')
+  const mockOracle = await oracleFactory.deploy()
+  await mockOracle.setData(toOracleValue('10000'), true)
+
+  const buttonTokenFactory = await ethers.getContractFactory('ButtonToken')
+  const buttonToken = await buttonTokenFactory.deploy()
+
+  buttonToken.initialize(weth.address, NAME, SYMBOL, mockOracle.address)
+
+  const routerFactory = await ethers.getContractFactory('ButtonTokenWethRouter')
+  const router = await routerFactory.deploy(weth.address)
+
+  return {
+    accounts,
+    buttonToken,
+    mockOracle,
+    weth,
+    router,
+  }
+}
+
+describe('ButtonTokenWethRouter', () => {
```

Comment:
add a check in all these tests which makes sure at no point does the router every own any button token, ETH, or WETH